### ISSUE TITLE
proposal: wait_until_present method with message argument

### ISSUE THREAD

**Issue by kubiksamek** [watir/watir#745]
Hi, is there any chance to add message argumet to `wait_until_present` method? My point is to avoid using `wait_until(.., message: "str", &:present?)` and to use `wait_until_present(.., message: "str")`. I would like to contribute to project, I just want to know if this change could pass. Or is there any different method I may looking for? Thank you.

**Comment by titusfortner** [comment]
We are actively encouraging people to use the approach you want to avoid. But looking at the code it does seem like we are splitting the difference oddly. I'm curious, what information you want to add to the error message?

**Comment by kubiksamek** [comment]
Error message? I just want to add `message` argument to `wait_until_present` method. That's all. But if you say that I should use `wait_until` method then fine :-).

**Comment by titusfortner** [comment]
The message argument is to override the default error message. I was wondering if there was a message you wanted to use or if you were just looking for parameters that could be added.

**Comment by kubiksamek** [comment]
Yes, I mean argument. Same as in `wait_until` method. So it would work like this:
```
def wait_until_present(deprecated_timeout = nil, timeout: nil, interval: nil, message: nil)
    if deprecated_timeout
        Watir.logger.deprecate "Using arguments for #wait_until_present", "keywords"
        timeout = deprecated_timeout
    end
    wait_until(timeout: timeout, interval: interval, message: message, &:present?)
end
```

**Comment by titusfortner** [comment]
First, @p0deje / @jkotests did we ever consider deprecating this method? I know we need `#wait_while_present` since it is subtly different than `#wait_while(&:present?)`, but the same isn't true for `#wait_until_present`. We got rid of `#wait_until_enabled`, I can't remember if there was a reason we kept this one; is it just too widely used?

As for the message, I can't think of a good use case for overriding the default message. That being said, the default message for this method has changed, and isn't as descriptive. If we were going to make a change, I'd advocate for changing the message back.

now:
>     "waiting for true condition on #{inspect}"
before:
>      "waiting for #{selector_string} to become present"

**Comment by p0deje** [comment]
I think we can deprecate `wait_until_present`, it was there for backwards-compatiblity IIRC.

**Comment by titusfortner** [comment]
@jkotests did you have any opinion on whether to deprecate `#wait_until_present` in favor of `wait_until(&:present?)`?

**Comment by jkotests** [comment]
@titusfortner, I have no preference.

**Comment by titusfortner** [comment]
Ok, there is now a specific use case for `#wait_until_present` that will require us to keep it: http://watir.com/guides/waiting/#wait-until-present-and-wait-while-present

@aliaCZEk Sorry for taking so long to evaluate this...

I actually pulled out the default message here: https://github.com/watir/watir/commit/a2549f47eefaef7ee0023c82b72b57bc31e77944#diff-0ac8030a39db381bf5dd122f0bfdd337L231

Which, I don't think I should have done, because we can provide more information than the default "Waiting for a true condition on..."

Also, there is no reason not to allow overriding message, even if there isn't an immediate use case for it.

So, yes, we can add back the default, and an override. @aliaCZEk are you interested in writing up that code?

**Comment by kubiksamek** [comment]
@titusfortner OK, why not :-)

**Comment by titusfortner** [comment]
This was fixed with #770
